File: depend/libcmb/python/tests/test3.py
#!/usr/bin/env python
from __future__ import print_function
from cmb import *
from sys import stderr

items = ["a", "b", "c", "d"]
choose = 2
nitems = len(items)

print("Enumerating choose-%s from %u:" % (choose, nitems), file = stderr)
config = CMB(size_min = choose, size_max = choose)
def afunc(items):
    print("\t%s" % (" ".join([x.decode('utf-8') for x in items])))
    return 0
cmb_callback(config, nitems, items, afunc)

File: depend/libcmb/tests/test3.c
/*-
 * Copyright (c) 2018-2026 Devin Teske <dteske@FreeBSD.org>
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */

#include <cmb.h>
#include <stdio.h>
#include <stdlib.h>

static int
afunc(struct cmb_config *config, uint64_t seq, uint32_t nitems, char *items[])
{
	int i;

	printf("\t");
	for (i = 0; i < nitems; i++)
		printf("%s%s", items[i], i < (nitems - 1) ? " " : "");
	printf("\n");

	return (0);
}

int
main(void)
{
	int nitems = 4;
	char *items[] = {"a", "b", "c", "d"};
	static struct cmb_config config = {
		.size_min = 2,
		.size_max = 2,
		.action = afunc,
	};

	printf("Enumerating choose-2 from %u:\n", nitems);
	(void)cmb(&config, nitems, items);

	return (EXIT_SUCCESS);
}
